fscache.go

```diff
@@ -175,105 +181,66 @@ func (c *cache) Clean() error {
 }
 
 type cachedFile struct {
-	fs   FileSystem
-	name string
-	cnt  int64
-	grp  sync.WaitGroup
-	w    io.WriteCloser
-	b    *broadcaster
+	stream *stream.Stream
+	cnt    int64
 }
 
 func (c *cache) newFile(name string) (*cachedFile, error) {
-	f, err := c.fs.Create(name)
+	s, err := stream.NewStream(name, c.fs)
 	if err != nil {
 		return nil, err
 	}
 	cf := &cachedFile{
-		fs:   c.fs,
-		name: f.Name(),
-		w:    f,
-		b:    newBroadcaster(),
+		stream: s,
 	}
-	cf.grp.Add(1)
 	atomic.AddInt64(&cf.cnt, 1)
 	return cf, nil
 }
 
 func (c *cache) oldFile(name string) *cachedFile {
-	b := newBroadcaster()
-	b.Close()
-	return &cachedFile{
-		fs:   c.fs,
-		name: name,
-		b:    b,
+	s, _ := stream.NewStream(name, c.fs)
```

Comment:
This causes problems, when the server is stopped then started again. What it is doing is erasing all the content in the file, which serves up EOFs. 
A potential solution is to make a `stream.OldStream` that calls `fs.Open` instead of `fs.Create`.